src/cs-gui/WebView.hpp

```diff
@@ -70,80 +76,10 @@ class CS_GUI_EXPORT WebView {
-      try {
-        callback(std::any_cast<A>(args[0]), std::any_cast<B>(args[1]));
-      } catch (std::bad_any_cast const& e) {
-        spdlog::error("Cannot execute javascript call '{}': {}", name, e.what());
-      }
-    });
-  }
-
-  /// See documentation above.
-  template <typename A, typename B, typename C>
-  void registerCallback(std::string const& name, std::function<void(A, B, C)> const& callback) {
-    registerJSCallbackImpl(name, [this, name, callback](std::vector<std::any> const& args) {
-      try {
-        callback(std::any_cast<A>(args[0]), std::any_cast<B>(args[1]), std::any_cast<C>(args[2]));
-      } catch (std::bad_any_cast const& e) {
-        spdlog::error("Cannot execute javascript call '{}': {}", name, e.what());
-      }
-    });
-  }
-
-  /// See documentation above.
-  template <typename A, typename B, typename C, typename D>
-  void registerCallback(std::string const& name, std::function<void(A, B, C, D)> const& callback) {
-    registerJSCallbackImpl(name, [this, name, callback](std::vector<std::any> const& args) {
-      try {
-        callback(std::any_cast<A>(args[0]), std::any_cast<B>(args[1]), std::any_cast<C>(args[2]),
-            std::any_cast<D>(args[3]));
-      } catch (std::bad_any_cast const& e) {
-        spdlog::error("Cannot execute javascript call '{}': {}", name, e.what());
-      }
-    });
-  }
-
-  /// See documentation above.
-  template <typename A, typename B, typename C, typename D, typename E>
-  void registerCallback(
-      std::string const& name, std::function<void(A, B, C, D, E)> const& callback) {
-    registerJSCallbackImpl(name, [this, name, callback](std::vector<std::any> const& args) {
-      try {
-        callback(std::any_cast<A>(args[0]), std::any_cast<B>(args[1]), std::any_cast<C>(args[2]),
-            std::any_cast<D>(args[3]), std::any_cast<E>(args[4]));
-      } catch (std::bad_any_cast const& e) {
-        spdlog::error("Cannot execute javascript call '{}': {}", name, e.what());
-      }
-    });
-  }
-
-  /// See documentation above.
-  template <typename A, typename B, typename C, typename D, typename E, typename F>
-  void registerCallback(
-      std::string const& name, std::function<void(A, B, C, D, E, F)> const& callback) {
-    registerJSCallbackImpl(name, [this, name, callback](std::vector<std::any> const& args) {
-      try {
-        callback(std::any_cast<A>(args[0]), std::any_cast<B>(args[1]), std::any_cast<C>(args[2]),
-            std::any_cast<D>(args[3]), std::any_cast<E>(args[4]), std::any_cast<F>(args[5]));
-      } catch (std::bad_any_cast const& e) {
-        spdlog::error("Cannot execute javascript call '{}': {}", name, e.what());
-      }
-    });
+  template <typename... Args>
+  void registerCallback(std::string const& name, std::function<void(Args...)> const& callback) {
+    assertJavaScriptTypes<Args...>();
+    registerCallbackWrapper(name, callback, std::index_sequence_for<Args...>{});
   }
```

Comment:
To avoid having to type `std::function` in the call site you can introduce a new template. Here is a minimal example:

```cpp
#include <iostream>

template <typename F, typename... Args>
void doSomething(F const &callback, Args &&... args) {
  callback(args...);
}

auto main() -> int {
  doSomething([](int a, double b) { std::cout << a << b << std::endl; }, 5, 0.1);
  doSomething([] { std::cout << "It works without std::function at the call site!"; });
  return 0;
}
```

Reply:
This doesn't work this way. The problem is, that in our example the Args are not passed to the function. This template parameter has to be deduced automatically which is not possible (afaik) if the entire lambda is passed in as a template parameter.